examples/stabilizer_code.py

```diff
@@ -98,47 +117,29 @@ def _GaussianElimination(M, min_row, max_row, min_col, max_col):
         self.n = n
         self.k = k
         self.r = r
-
-        def _BuildByCode(mat):
-            out = []
-            n = mat.shape[1] // 2
-            for i in range(mat.shape[0]):
-                ps = ''
-                for j in range(n):
-                    if mat[i, j] == 0 and mat[i, j + n] == 0:
-                        ps += 'I'
-                    elif mat[i, j] == 1 and mat[i, j + n] == 0:
-                        ps += 'X'
-                    elif mat[i, j] == 0 and mat[i, j + n] == 1:
-                        ps += 'Z'
-                    else:
-                        ps += 'Y'
-                out.append(ps)
-            return out
-
         self.M = _BuildByCode(M)
         self.X = _BuildByCode(X)
         self.Z = _BuildByCode(Z)
         self.syndromes_to_corrections = {}
 
         for qid in range(self.n):
-            for op in ['X', 'Z']:
-                syndrome = [
+            for op in allowed_errors:
+                syndrome = tuple(
                     1 if self.M[r][qid] == 'I' or self.M[r][qid] == op else -1
-                    for r in range(self.r)
-                ]
-                self.syndromes_to_corrections[tuple(syndrome)] = (op, qid)
+                    for r in range(self.n - self.k)
+                )
+                self.syndromes_to_corrections[syndrome] = (op, qid)
+
+    def encode(self, qubits):
```

Comment:
As with the external functions, docstrings and type annotations on these methods (`encode`, `correct`, and `decode`) would make this example easier to read.